pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -125,6 +134,8 @@ public abstract class BaseBrokerRequestHandler implements BrokerRequestHandler {
   private final int _defaultHllLog2m;
   private final boolean _enableQueryLimitOverride;
   private final boolean _enableDistinctCountBitmapOverride;
+  private final Map<Long, QueryRoutingTable> _queriesById = new ConcurrentHashMap<>();
```

Comment:
(minor) set it to `null` when query cancellation is disabled

Reply:
the kv pair is removed in a finally{} block when query ends